### PR TITLE
3.0 `azurerm_policy_remediation` split into `azurerm_management_group_policy_remediation`, `azurerm_resource_policy_remediation`, `azurerm_resource_group_policy_remediation` and `azurerm_subscription_policy_remediation`

### PR DESCRIPTION
**Pull request by mbfrahry**
The `azurerm_policy_remediation` resource will be removed in favour of the `azurerm_management_group_policy_remediation`, `azurerm_resource_policy_remediation`, `azurerm_resource_group_policy_remediation` and `azurerm_subscription_policy_remediation` resources.

image

### CODE REVIEW

File: internal/services/policy/remediation_management_group.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validate.PolicyAssignmentID,
+			},
+
+			"location_filters": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+				},
```
Comment (katbyte) [inline internal/services/policy/remediation_management_group.go:71]: should we validate this? at least not empty?

File: internal/services/policy/remediation_management_group.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+			"policy_definition_id": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				// TODO: remove this suppression when github issue https://github.com/Azure/azure-rest-api-specs/issues/8353 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validate.PolicyDefinitionID,
+			},
+
+			"resource_discovery_mode": {
```
Comment (katbyte) [inline internal/services/policy/remediation_management_group.go:82]: is resource really needed? 
```suggestion
			"discovery_mode": {
```

**Review by katbyte — APPROVED** [review]
aside from 2 comments on schema LGTM ⚔️

File: internal/services/policy/remediation_management_group.go (hunk 1 above)
Comment (mbfrahry) [reply internal/services/policy/remediation_management_group.go:71]: Ohhh, good call!

File: internal/services/policy/remediation_management_group.go (hunk 2 above)
Comment (mbfrahry) [reply internal/services/policy/remediation_management_group.go:85]: I believe so! It's a specific callout for how we'll find which resources to remediate under the umbrella of each type of remediation policy.

**Pull request merged**